app/controllers/authenticate_controller.rb

```diff
@@ -222,16 +211,9 @@ def handle_authentication_error(err)
     when Errors::Authentication::Jwt::TokenExpired
       raise Unauthorized.new(err.message, true)
 
-    when Errors::Authentication::OAuth::ProviderDiscoveryTimeout
-      raise GatewayTimeout
-
     when Errors::Util::ConcurrencyLimitReachedBeforeCacheInitialization
       raise ServiceUnavailable
 
-    when Errors::Authentication::OAuth::ProviderDiscoveryFailed,
-      Errors::Authentication::OAuth::FetchProviderKeysFailed
-      raise BadGateway
```

Comment:
@sashaCher can you please elaborate on this change? why are we not returning a gateway timeout error in this case?
![image](https://user-images.githubusercontent.com/29940926/129565901-5a32709e-eb95-4674-ab4b-ebb6e7656b3c.png)

this seems a much better error than the generic 401

Reply:
Updated PR description